python/src/spark_rapids_ml/params.py

```diff
@@ -163,6 +176,26 @@ def num_workers(self) -> int:
     def num_workers(self, value: int) -> None:
         self._num_workers = value
 
+    def copy(self: P, extra: Optional["ParamMap"] = None) -> P:
+        # override this function to update cuml_params if possible
+        instance: P = super().copy(extra)
```

Comment:
Not sure we want `super().copy()` here, since Spark generally uses `self.copy()`.

Reply:
Yeah, super().copy() is a convenient way to avoid handling extra parameters manually.